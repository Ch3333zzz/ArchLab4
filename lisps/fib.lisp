(defun fib (n)
  (if (<= n 1)
      n
      (+ (call fib (- n 1)) (call fib (- n 2)))))

(setq res (call fib 6))
(print res)
