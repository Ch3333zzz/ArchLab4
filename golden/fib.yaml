name: fib_recursive
config: 
  lenient_log: true
source:
  language: lisp
  code: |
    (defun fib (n)
      (if (<= n 1)
          n
          (+ (call fib (- n 1)) (call fib (- n 2)))))

    (print (call fib 6))

input_schedule: []
expect:
  output: '8'
  state: stopped
  error: false
  out_code: !!binary |
    BgAACgAAADwUAAAoAAAARgAAAAEBAAFSAAAAUAAAADwBAAAKBAAAWzQAAB8AAABQbAAAHgAAAFAAAAA8AQAACgIAAFoAAAA8FAAAKAAAADwAAABQAAAAPAIAAAoCAABaAAAAPBQAACgBAABacAAAHgAAAFMAAAAp
  out_code_hex: |
    0 - 0600000A - LOAD_IMM 6
    4 - 0000003C - PUSH_IMM 0
    8 - 14000028 - CALL 20
    12 - 00000046 - PRINT
    16 - 00000001 - HALT
    20 - 01000152 - ENTER 65537
    24 - 00000050 - LOAD_LOCAL 0
    28 - 0000003C - PUSH_IMM 0
    32 - 0100000A - LOAD_IMM 1
    36 - 0400005B - CMP_POP 4
    40 - 3400001F - BEQZ 52
    44 - 00000050 - LOAD_LOCAL 0
    48 - 6C00001E - JMP 108
    52 - 00000050 - LOAD_LOCAL 0
    56 - 0000003C - PUSH_IMM 0
    60 - 0100000A - LOAD_IMM 1
    64 - 0200005A - BINOP_POP 2
    68 - 0000003C - PUSH_IMM 0
    72 - 14000028 - CALL 20
    76 - 0000003C - PUSH_IMM 0
    80 - 00000050 - LOAD_LOCAL 0
    84 - 0000003C - PUSH_IMM 0
    88 - 0200000A - LOAD_IMM 2
    92 - 0200005A - BINOP_POP 2
    96 - 0000003C - PUSH_IMM 0
    100 - 14000028 - CALL 20
    104 - 0100005A - BINOP_POP 1
    108 - 7000001E - JMP 112
    112 - 00000053 - LEAVE 0
    116 - 00000029 - RET
  out_log: |
    DEBUG root:processor.py:197 Datapath: heap_ptr initialized to word 1054
    DEBUG root:processor.py:616 No more scheduled input and mmio_in_buffer empty -> input closed (EOF)
    DEBUG root:processor.py:1095 PRINT: numeric -> 8
    DEBUG root:processor.py:320 [MMIO OUT] wrote 56 -> char: 8
    DEBUG root:processor.py:725 HALT encountered









